### ISSUE TITLE
[Issue]: No longer installs on Fedora 36, bump version?

### ISSUE THREAD

**Issue by Whayme** [jellyfin/jellyfin#7471]
### Please describe your bug

Installing the latest unstable build on Fedora 36 Prerelease, it returns:

```Error: 
 Problem: conflicting requests
  - nothing provides liblttng-ust.so.0()(64bit) needed by jellyfin-server-20220317.2-1.fc33.x86_64
```
As the Fedora release targets 33, I imagine targeting 35 or 36 is in order.

### Jellyfin Version

Other

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by crobibero** [comment]
Feel free to make a PR with the requited changes. Fedora/rpm support is community maintained.

**Comment by mihawk90** [comment]
> on Fedora 36 Prerelease

> `fc33.x86_64`

Well you're trying to install an F33 package on F36 so that's sort of expected...

Anyway, certainly related to #7504

If I wanted to look into this, where would I even start? On what infrastructure are RPMs currently built?

**Comment by crobibero** [comment]
Likely here: https://github.com/jellyfin/jellyfin/tree/master/fedora

**Comment by mihawk90** [comment]
Thank you, seems I missed that because I was mixing up the repo with the jellyfin-web one :facepalm: 

Anyway, it seems this has already been done in https://github.com/jellyfin/jellyfin/commit/543b0127b363d3fd371eb08166633ba6e0d2f9bd:
https://github.com/jellyfin/jellyfin/blob/49d5fdb33fc9792147c1df03e1d1b051e6b7ec79/fedora/Makefile#L12-L14

Except that last line seems to be a typo and is probably meant to be 36 instead. I can PR that though.
My guess is that on next release it will probably be built on those 3 already.

I'm just wondering why all the current [alpha/beta downloads](https://repo.jellyfin.org/releases/server/fedora/versions/stable-pre/server/10.8.0-beta1/) are still on F33 then when this code has been in for 4 months already :thinking:

**Comment by mihawk90** [comment]
Hm so I looked into this briefly on the go...

This file is from the `lttng-ust` package. The F36 package was already built during the mass-rebuild during branching back in January:

https://koji.fedoraproject.org/koji/buildinfo?buildID=1888606

So the package is available, however the file we're looking for is not:
https://packages.fedoraproject.org/pkgs/lttng-ust/lttng-ust/fedora-36.html#provides

Notice the package provides `liblttng-ust.so.1()(64bit)`, but not the `.0`.

I'll have to look into this further when I'm not half asleep.

**Comment by 24fpsDaVinci** [comment]
> Notice the package provides `liblttng-ust.so.1()(64bit)`, but not the `.0`.
this is the main problem trying to install on fedora 36

**Comment by mihawk90** [comment]
Currently trying to figure out if that is intended.

If the Ubuntu package is any indication I would say it is not:
https://packages.ubuntu.com/impish/amd64/liblttng-ust0/filelist

```
[...]
/usr/lib/x86_64-linux-gnu/liblttng-ust.so.0
/usr/lib/x86_64-linux-gnu/liblttng-ust.so.0.0.0
[...]
```

On the other hand, the Arch package:
https://archlinux.org/packages/community/x86_64/lttng-ust/
```
[...]
usr/lib/liblttng-ust.so
usr/lib/liblttng-ust.so.1
usr/lib/liblttng-ust.so.1.0.0
[...]
```

So we have precedent for both. I'd just be curious what exactly links to the `.0` file, whether it's Jellyfin itself or a component of Dotnet.
Should be noted though that Arch and Fedora 36 are both on 2.13.2 and Ubuntu is on 2.12.1.

This makes me think that the culprit here is actually Dotnet which might be built against an older `lttng-ust`.
Speaking of: It appears that Dotnet6.0 has not actually been built against Fedora 36 yet. The corresponding repo doesn't have an f36 branch:
https://copr-dist-git.fedorainfracloud.org/cgit/@dotnet-sig/dotnet-preview/dotnet6.0.git

And although there is a Fedora 36 listed on the COPR repo we import, it hasn't been built yet:

Chroot Name | Dist Git Source | Build Time | Logs | State
-- | -- | -- | -- | --
fedora-36-x86_64 | 2ad0f2d | 2 hours | RPM build has not started yet | forked

TL;DR: It appears we can't do much about this at this time and can just wait for a new build to roll around until F36 hits release.

**Comment by mihawk90** [comment]
Man... this thing is really bothering me more then it should :P 

Did some more digging today... My task today was to find out what actually even links to that .so to be able to file a bug report in the right place.
Since RPM's automatic dependency detection is borderline black magic it took me a bit to dig through, but I did find it.

TLDR: Seems the packaging process packages a different .so file then is provided in the `dotnet-runtime-6.0` package. Not sure why.

---

For the interested, there are two ways.

The RPM way:
The way RPM auto detects dependencies is a script whose location you can find using this:
```
% rpm --eval "%__find_requires"
/usr/lib/rpm/find-requires
```
This can technically be modified, but by default it just contains this:
```
#!/usr/bin/sh

/usr/lib/rpm/rpmdeps --define="_use_internal_dependency_generator 1" --requires
```
So using that binary we can determine which file is actually linking to the required .so. For options see `man rpmdeps`, but:
```
% /usr/lib/rpm/rpmdeps --alldeps ./*.so* jellyfin createdump
  8 ./libcoreclrtraceptprovider.so
        P libcoreclrtraceptprovider.so()(64bit)
        R libc.so.6(GLIBC_2.4)(64bit)
        R libc.so.6(GLIBC_2.2.5)(64bit)
>>>     R liblttng-ust.so.0()(64bit)
        R libstdc++.so.6()(64bit)
        R libm.so.6()(64bit)
        R libgcc_s.so.1()(64bit)
        R libc.so.6()(64bit)
        R rtld(GNU_HASH)
```
And there we got our culprit.

The LDD way:
```
% ldd ./*.so* jellyfin createdump
./libcoreclrtraceptprovider.so:
        linux-vdso.so.1 (0x00007ffcb270b000)
>>>     liblttng-ust.so.0 => not found
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f8aa95dc000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f8aa94fe000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f8aa94de000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f8aa92dc000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8aa98e3000)
```

---

From what I can tell `libcoreclrtraceptprovider.so` is a component of dotnet, why it links against the wrong .so is beyond me though.

My first thought was that for some reason the package it's coming from (`dotnet-runtime-6.0`) was compiled using the wrong `lttng-ust-devel` version. So I unpacked RPM and checked its dependencies on this particular .so:

```
% /usr/lib/rpm/rpmdeps --alldeps libcoreclrtraceptprovider.so
  0 libcoreclrtraceptprovider.so
        P libcoreclrtraceptprovider.so()(64bit)
        R libc.so.6(GLIBC_2.4)(64bit)
        R libc.so.6(GLIBC_2.34)(64bit)
        R libc.so.6(GLIBC_2.2.5)(64bit)
        R libstdc++.so.6(CXXABI_1.3)(64bit)
        R libstdc++.so.6(GLIBCXX_3.4)(64bit)
>>>     R liblttng-ust.so.1()(64bit)
        R libstdc++.so.6()(64bit)
        R libm.so.6()(64bit)
        R libgcc_s.so.1()(64bit)
        R libc.so.6()(64bit)
        R rtld(GNU_HASH)
```
So this one does in fact link against the correct .so.

I am at a loss how the packaging process changes the link target however, or more accurately, uses a different .so file then provided in the runtime package. I don't know why and it makes little sense to me.
I have absolute no experience with .NET but my guess would be it's doing some internal versioning and we need to target the 103 or 104 runtime specifically.

**Comment by mihawk90** [comment]
Day 6541729. Days are getting shorter, I'm still digging~

So today I scoured through the dotnet bugtracker and checked if there's maybe a known issue with what we're experiencing. And it turns out there is:
https://github.com/dotnet/sdk/issues/22220

So I tried a build without specifying the `--output` parameter. And while the package didn't finish (because I was lazy and didn't care about updating the package list properly), we can at least see the resultant files in the mock build directory. Unfortunately, it's still getting me the wrong dotnet .so files :( :
```
[tarulia@fedora]/var/lib/mock/fedora-36-x86_64/root/builddir/build/BUILD/jellyfin-server-10.8.0~beta2/Jellyfin.Server/bin/Release/net6.0/fedora-x64/publish% ldd --verbose libcoreclrtraceptprovider.so 
        linux-vdso.so.1 (0x00007ffd209fa000)
>>>     liblttng-ust.so.0 => not found
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fa2280f2000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fa228014000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fa227ff4000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fa227df2000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa2283f9000)
```

I'm starting to run out of ideas here...

**Comment by mihawk90** [comment]
> The only option I can think of right now is super mega hacky, ugly, and just pure brute force method.

So I tried this (and I feel bad and dirty for it), and it... works*?

*works as in: it installs, jellyfin starts and does its thing, but I have no way of testing every single feature. And I also don't know what parts even actually use those libs so I can't do targeted testing either.

Anyway, if anyone wants to test them, here you go:
https://github.com/mihawk90/jellyfin/releases/tag/v10.8.0-beta2

Keep in mind these are super WIP for my PRs and technically including a library like that is really not encouraged, but hey what you gonna do. Also keep in mind this in no way means that the final release will actually be done like this (hopefully dotnet gets their s**t together before then).

**Comment by SR-Lut3t1um** [comment]
Well, this really does sound hacky. Can't we simply use this workaround?: 

https://github.com/intel-iot-devkit/meta-iot-cloud/pull/106#issuecomment-1017961611

Seams to be the cleaner solution.

**Comment by mihawk90** [comment]
> Can't we simply use this workaround?

TBH I haven't gotten around to looking for workarounds yet. The test build yesterday was just a PoC done in like 5 minutes (including compile) to see if it was even an option should we have to go this route.

From that comment:

> On Fedora, we would [filter out the automatic dependency](https://docs.fedoraproject.org/en-US/packaging-guidelines/AutoProvidesAndRequiresFiltering/#_filtering_provides_and_requires_after_scanning).

Well funny enough automatic dependencies are already supposed to be disabled:

https://github.com/jellyfin/jellyfin/blob/3fc8652c265fe0b9c6876f0b050aed9ef31b187c/fedora/jellyfin.spec#L34

But I guess that directive doesn't work anymore (if it was ever correct, since the [AutoReqProv draft](https://fedoraproject.org/wiki/AutoReqProv_(draft)) only talked about the filter macros, not a general disabling).

> And if someone really needs `libcoreclrtraceptprovider.so`, well, they will have to get a working `liblttng-ust.so.0` anyway before they can use any .NET version with it.

That was also what I was wondering yesterday and asked in the Dev channel, but unfortunately noone answered on it.
I don't know whether `libcoreclrtraceptprovider.so` is even actually being used or if it is just an artifact of .NET's build process.

The option described in the linked comment is one way to do it. Another would be removing `libcoreclrtraceptprovider.so` from the package entirely, but for that I'd need to know wether it is being used first.

**Comment by mihawk90** [comment]
OK so... I got a build running that excludes the dependency. I still need to do the conditionals (since we don't want to exclude the dependency on Fedora versions that don't need it). And here is my dilemma:

I would like to target only affected versions, which as of right now is only F36. However, it seems `lttng-ust` isn't releasing all that often. When we look at the [lttng-ust repo](https://github.com/lttng/lttng-ust) they have a new tag every couple months, but the real question is when will we go back to a `liblttng-ust.so.0`, if ever?

Any thoughts or strong feelings on this?

**Comment by haysberg** [comment]
Wondering the exact same thing. I'm wondering if leaving the packet as broken will break my installation as the library file will still be there in the exact same path.

**Comment by mihawk90** [comment]
> With Fedora 36 beeing released today, what is the current state of your PR? @mihawk90

I put it up for review 2 weeks ago, so it's up to the maintainers now.

You can compile locally using the build script if you need it.

Regardless of the state of the PR, I imagine there won't be an official build for the new versions before 10.8 releases anyway.

**Comment by haysberg** [comment]
Thank you for your instructions. I will wait for the 10.8 release however (whenever that happens), for the sake of keeping my system coherent.

**Comment by haysberg** [comment]
Well looking at the changelog for the latest beta, sounds like we're out of luck for this release boys.

**Comment by SR-Lut3t1um** [comment]
Yeah seems like it. The stable-backport label also got removed :/

**Comment by mihawk90** [comment]
The PR is against the Beta branch, as was requested, so it's not over until 10.8 is actually released.

**Comment by haysberg** [comment]
🤞

**Comment by EvansM4** [comment]
Has this been fixed yet?

**Comment by SR-Lut3t1um** [comment]
> The PR is against the Beta branch, as was requested, so it's not over until 10.8 is actually released.

Well well well :/

**Comment by mooninite** [comment]
Problem:
* Fedora 36 upgraded lttng-ust to 2.13.0, which bumped the SONAME from .0 to .1
* Fedora 36 dotnet-runtime-6.0 is compiled against lttng-ust 2.13.0
* The dotnet publish command pulls in microsoft.netcore.app.runtime.linux-x64 from nuget and uses its binaries instead of the host runtime binaries in the publish directory output. The upstream nuget package is compiled against an older lttng-ust library with an SONAME of .0. The 'libcoreclrtraceptprovider.so' runtime binary is what has a dependency on the older lttng-ust.

Solution(s):
* Ask dotnet upstream to offer a runtime nuget compiled against lttng-ust 2.13 or newer
* Remove the 'libcoreclrtraceptprovider.so' binary from the Fedora package. It doesn't seem to be in use by Jellyfin. I ran a 10.8.0 server and it can play video without that binary being present. I don't think many/any Fedora users will use the trace functionality that binary offers.

**Comment by mihawk90** [comment]
If you had read the discussion you'd know this has all been explored already, and also fixed in the aforementioned PR.

**Comment by mooninite** [comment]
Except your fix is not a fix. It ships a binary that won't work. Best to not ship it at all.

**Comment by mihawk90** [comment]
That's for the maintainers to decide.

And besides it can be *made* to work, just need to install the older lttng-ust separately.

**Comment by 6jarjar6** [comment]
Got it to install by installing the dependency from here https://fedora.pkgs.org/35/fedora-x86_64/lttng-ust-2.12.2-5.fc35.x86_64.rpm.html

**Comment by mihawk90** [comment]
Probably, I was just waiting for stale-bot to close this but for some reason it's not looking at this issue 🤷
